pytorch_lightning/plugins/training_type/ddp_spawn.py

```diff
@@ -82,7 +76,7 @@ def distributed_sampler_kwargs(self):
     def setup(self, model):
         self._model = model
 
-        os.environ["MASTER_PORT"] = os.environ.get("MASTER_PORT", str(find_free_network_port()))
+        os.environ["MASTER_PORT"] = str(self.cluster_environment.master_port())
```

Comment:
we don't need `str(...)` here right? the method returns a str

Reply:
no, the port is a number (int) and writing it into the `os.environ` we need to convert it back to str. In the future, we may want to clean this up so that we don't have to write to the original environment and only do it for the subprocess. For now, this part of the code is just a refactor.

Reply:
In the future, I think it should the cluster environment's job to set the var, if necessary